lib/vdsm/storage/lvm.py

```diff
@@ -1417,12 +1420,11 @@ def getVGbyUUID(vgUUID):
 
 
 def getLV(vgName, lvName=None):
-    lv = _lvminfo.getLv(vgName, lvName)
-    # getLV() should not return None
-    if not lv:
-        raise se.LogicalVolumeDoesNotExistError("%s/%s" % (vgName, lvName))
-    else:
-        return lv
+    """
+    Return LV named tuple. Raise se.LogicalVolumeDoesNotExistError if the
+    LV does not exist.
+    """
+    return _lvminfo.getLv(vgName, lvName)
```

Comment:
This is not like getVG - when called with lvName=None, this returns all the lvs
in the vg, and raising se.LogicalVolumeDoesNotExistError does not make sense
since the caller did not ask to get a certain lv.

If we want to unify this we need to require lvName != None here and handle only the
case of returning single lv, and add another function getAllLVs(vgName) for
returning list of all lvs - which will never raise if the list is empty.

We also need to change LVMCache.getLv() to always require lv name, and add another methods like LVMCache.getAllLVs().

Lets first make the API sane before we try to unify error handling.

Reply:
If we want to make the API consistent with VGs (which sounds good to me) we definitively need to add those extra methods. Also, it probably will make it easier for the reader to follow the code.

But the current API already does what you describe, only raise if `lvName != None`. This is handled by the `LVMCache.getLv()`. It will not raise if the list is empty, and actually that behavioral change triggered the changes in the `lvm_test` and `blockSD`. This is reflected in the commit message. However, it is not reflected correctly in the docstring of this function, unfortunately.

Reply:
The current API may work but I don't want to invest time in improving
it.